tests/components/vesync/test_fan.py

```diff
@@ -0,0 +1,50 @@
+"""Tests for the fan module."""
+import pytest
+import requests_mock
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.fan import DOMAIN as FAN_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from .common import ALL_DEVICE_NAMES, mock_devices_response
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize("device_name", ALL_DEVICE_NAMES)
+async def test_fan_state(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    config_entry: MockConfigEntry,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    requests_mock: requests_mock.Mocker,
+    device_name: str,
+) -> None:
+    """Test the resulting setup state is as expected for the platform."""
+
+    # Configure the API devices call for device_name
+    mock_devices_response(requests_mock, device_name)
+
+    # setup platform - only including the named device
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Check device registry
+    devices = dr.async_entries_for_config_entry(device_registry, config_entry.entry_id)
+    assert devices == snapshot(name="devices")
+
+    # Check entity registry
+    entities = [
+        entity
+        for entity in er.async_entries_for_config_entry(
+            entity_registry, config_entry.entry_id
+        )
+        if entity.domain == FAN_DOMAIN
+    ]
+    assert entities == snapshot(name="entities")
+
+    # Check states
+    for entity in entities:
+        assert hass.states.get(entity.entity_id) == snapshot(name=entity.entity_id)
```

Comment:
My instincts want me to extract this code block as a common method.  There is only one difference between all four function calls and it is the domain value on line 44.

I've left it unrefactored to ensure that it first met the desired feedback.  refactoring could occur as a change on this PR or in a future PR, I'll leave the decision to the reviewer.

Reply:
You're right - I think it might make sense to introduce a global helper for this: #90775

Reply:
Based on internal discussions, having a helper makes the tests harder to follow, so it is recommended to keep this as separate methods inside each platform.